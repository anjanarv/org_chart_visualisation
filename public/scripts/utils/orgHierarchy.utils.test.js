const peopleInGivenCodingChallenge = require("../../data/peopleInGivenCodingChallenge.json");
const peopleWithNoId = require("../../data/peopleWithNoId.json");
const { validateJSON } = require("./orgHierarchy.utils");

test("validating json having valid data", () => {
  let { validPersons, invalidPersons } = validateJSON(
    peopleInGivenCodingChallenge
  );
  expect(validPersons).toMatchSnapshot();
  expect(invalidPersons).toMatchSnapshot();
});

test("validating json having invalid data", () => {
  let { validPersons, invalidPersons } = validateJSON(peopleWithNoId);

  expect(validPersons).toMatchSnapshot();
  expect(invalidPersons).toMatchSnapshot();
});

test("validating json having empty data", () => {
  let { validPersons, invalidPersons } = validateJSON([]);

  expect(validPersons).toStrictEqual([]);
  expect(invalidPersons).toStrictEqual([]);
});
