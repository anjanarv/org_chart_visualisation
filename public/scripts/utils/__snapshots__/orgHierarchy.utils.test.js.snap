// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validating json having invalid data 1`] = `Array []`;

exports[`validating json having invalid data 2`] = `
Array [
  Object {
    "managerId": "150",
    "name": "Alan",
  },
  Object {
    "managerId": "100",
    "name": "Martin",
  },
  Object {
    "name": "Jamie",
  },
]
`;

exports[`validating json having valid data 1`] = `
Array [
  Object {
    "id": "100",
    "managerId": "150",
    "name": "Alan",
  },
  Object {
    "id": "150",
    "name": "Jamie",
  },
  Object {
    "id": "400",
    "managerId": "150",
    "name": "Steve",
  },
  Object {
    "id": "190",
    "managerId": "400",
    "name": "David",
  },
]
`;

exports[`validating json having valid data 2`] = `
Array [
  Object {
    "managerId": "100",
    "name": "Martin",
  },
  Object {
    "managerId": "100",
    "name": "Alex",
  },
]
`;
